.pipelines/templates/PackageBuild.yml

```diff
@@ -226,6 +231,14 @@ steps:
         max_cascading_rebuilds_arg="MAX_CASCADING_REBUILDS=${{ parameters.maxCascadingRebuilds }}"
       fi
 
+      if [[ ${{ parameters.dailyBuildID }} == "lkg" ]]; then
+        daily_build_id_arg="DAILY_BUILD_ID=${{ parameters.dailyBuildID }}"
+        daily_build_update_manifests_arg="DAILY_BUILD_ID_UPDATE_MANIFESTS=n"
+      elif [[ -n "${{ parameters.dailyBuildID }}" ]]; then
+        daily_build_id_arg=""
+        daily_build_update_manifests_arg=""
+      fi
```

Comment:
We're getting closer but this is not exactly it. The `PackageBuild.yml` template does not make any guesses as to what the caller wants to do, so hard-coding `DAILY_BUILD_ID_UPDATE_MANIFESTS=n` is not the way to go. It would be confusing behaviour, if someone used that template to build packages from a dev branch and with `dailyBuildID=lkg` and got a build error, because the manifests were not updated. From the point of view of this template, these two arguments are independent and should be treated as such. The caller is the one, who knows how they want to build the packages and will thus set them accordingly.

Reply:
The thing is that `DAILY_BUILD_ID_UPDATE_MANIFESTS` depends on `DAILY_BUILD_ID` being set. So someone passing down `dailyBuildID` as empty string parameter to this template and then something like `dailyBuildUpdateManifests` parameter as `true`/`y` would not make sense and result in no different behavior than just passing alone `dailyBuildID` parameter as empty string

Reply:
Updated